src/main/java/life/qbic/model/download/OpenbisConnector.java

```diff
@@ -41,15 +63,122 @@ public OpenbisConnector(
     } else {
       applicationServer = null;
     }
+
+     */
+  }
+
+  public OpenbisConnector(OpenBIS authentication) {
+    this.openBIS = authentication;
   }
 
   public List<String> getSpaces() {
     SpaceSearchCriteria criteria = new SpaceSearchCriteria();
     SpaceFetchOptions options = new SpaceFetchOptions();
-    return applicationServer.searchSpaces(sessionToken, criteria, options).getObjects()
+    return openBIS.searchSpaces(criteria, options).getObjects()
         .stream().map(Space::getCode).collect(Collectors.toList());
   }
 
+  public DataSetPermId registerDataset(Path uploadPath, String experimentID, List<String> parentCodes) {
+    final String uploadId = openBIS.uploadFileWorkspaceDSS(uploadPath);
+
+    final UploadedDataSetCreation creation = new UploadedDataSetCreation();
+    creation.setUploadId(uploadId);
+    creation.setExperimentId(new ExperimentIdentifier(experimentID));
+    creation.setParentIds(parentCodes.stream().map(DataSetPermId::new).collect(
+        Collectors.toList()));
+    creation.setTypeId(new EntityTypePermId("UNKNOWN", EntityKind.DATA_SET));
```

Comment:
NIT: Do you think i'd make sense to specify these values as final static variables with a descriptive name? For me it's unclear what the purpose of UNKOWN is in this context 🤔

Reply:
good point, at this point it's a proof of concept and "UNKNOWN" is the dataset type we can be sure of existing in the system